modules/sdk/src/main/scala/org/tessellation/sdk/http/routes/SnapshotRoutes.scala

```diff
@@ -64,6 +65,17 @@ final case class SnapshotRoutes[F[_]: Async: KryoSerializer, S <: Snapshot[_, _]
         }
       }
 
+    case GET -> Root / "hash" / SnapshotOrdinalVar(ordinal) =>
+      import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
+
+      snapshotStorage
+        .get(ordinal)
+        .flatMap(_.traverse(_.toHashed.map(_.hash)))
```

Comment:
I think it makes more sense to keep that logic in the `snapshotStorage`. I mean additional method there e.g. `getHash(ordinal: SnapshotOrdinal): F[Option[Hash]]` that fetches the snapshot and hashes it, there could also be an optimization there where in the case the hash is in the `ordinalCache` we don't need to rehash the snapshot.

In the future I would consider if having a cache of ordinal to hash mapping would make sense (to not hash a snapshot on every request which could be computationally costly), maybe we could figure out a way to check what file does the ordinal link to and get the hash that way (snapshots are saved on the disk under hash name and a link by ordinal is created to them). But let's not optimize prematurely and get it working first.